src/server/core/webserver.ts

```diff
@@ -96,6 +98,9 @@ export class WebServer extends EventEmitter implements BaseKrasServer {
     this.server = ssl ? createHttpsServer(ssl, this.app) : createHttpServer(this.app);
     this.wsOptions = typeof config.ws === 'object' ? config.ws : undefined;
     this.ws = !!config.ws;
+    this.permitHeaders = config.injectors.proxy.active
+      ? (config.injectors.proxy as ProxyInjectorConfig).permitHeaders
```

Comment:
Unfortunately, this cannot be done like that

1. There is no guarantee that `proxy` is there
2. Even if it is - the webserver module should not know about that injector specific configuration

Maybe a better solution would be to use the `access-control-request-headers` header of the request to dynamically set the allowed headers. This way, there is no config for proxy necessary, and it would also work with the other injectors without configuration of the proxy (or any other) injector.

Reply:
Sounds better, I will try and come back to update. :+1: